Topics/starting-a-data-project.md

```diff
@@ -0,0 +1,254 @@
+
+Once you've walked through the above resources, you'll be equipped with best practices to ensure that your code will be highly reproducible, and again, people will be grateful to you! 
+
+**Especially, the future yourself will be happy!!**
+
+**Must read:**
+- [**Ten Simple Rules for Reproducible Computational Research**](http://dx.plos.org/10.1371/journal.pcbi.1003285)
+- [**Ten Simple Rules for Reproducible Research in Jupyter Notebooks**](https://arxiv.org/ftp/arxiv/papers/1810/1810.08055.pdf)
+
+### Versioning
+
+In order to make reproducible projects, and also for peace in the heart of every programmer, learn to use [**Git**](https://git-scm.com/)!
+
+Git is a versioning system that allows you to always have under control every change in your code, be able to go back, and be sure that your code will never be lost!
+
+Git is defined as **Distributed Version Control System**: What does it mean?
+
+From [this article](https://www.freecodecamp.org/news/what-is-git-and-how-to-use-it-c341b049ae61/):
+
+- Control System: This basically means that Git is a content tracker. So Git can be used to store content — it is mostly used to store code due to the other features it provides.
+
+- Version Control System: The code which is stored in Git keeps changing as more code is added. Also, many developers can add code in parallel. So Version Control System helps in handling this by maintaining a history of what changes have happened. Also, Git provides features like branches and merges, which I will be covering later.
+
+- Distributed Version Control System: Git has a remote repository that is stored in a server and a local repository that is stored in the computer of each developer. This means that the code is not just stored in a central server, but the full copy of the code is present in all the developers’ computers. Git is a Distributed Version Control System since the code is present in every developer’s computer. I will explain the concept of remote and local repositories later in this article.
+
+_Any existing software project that is not under version control is considered a dead project, and the responsible developers are considered crazy._
+
+Data Science projects (which make heavy use of software) are no different, indeed! 
+
+They also have the _additional problem of data versioning_, which is the raw material on which you work most. 
+
+Always having the versions of the data, from raw (just collected) to clean, keeping every intermediate processing phase, is perhaps the most important [best practice](https://medium.com/thelaunchpad/retracing-your-steps-in-machine-learning-ml-versioning-74d19a66bd08) when doing a Data Science project.
+
+[**Here**](https://rogerdudler.github.io/git-guide/index.html) you can find a simple guide to Git. Learn it, it's freaking worth (and necessary).
+
+Documenting your work with Git is crucial: read [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/).
```

Comment:
👍 

I'll suggest adding some Git/GitHub related resources i.e. Tutorials, videos, blog posts etc... to help learn the basics and more!